client/src/pages/StadiumProfile.tsx

```diff
@@ -1,10 +1,63 @@
 import { Box } from '@mui/system';
-import ImageSlider from '../components/stadiumProfile/ImageSlider';
+import { ReactElement, useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import axios from 'axios';
+import {
+  BioSection,
+  ImageSlider,
+  ProfileCalender,
+} from '../components/stadiumProfile';
+import { UserData } from '../interfaces/StadiumProfile';
 
-const StadiumProfile = () => {
+interface errorI {
+  response: { status: number; data: object };
+}
+const StadiumProfile = (): ReactElement => {
+  const [userData, setUserData] = useState<UserData | null>(null);
+  const [gallery, setGallery] = useState([
+    {
+      id: 0,
+      image: '',
+    },
+  ]);
+
+  const navigate = useNavigate();
+
+  const { id } = useParams();
+  const fetchProfileData = async (stadiumId: string) => {
+    try {
+      const { data } = await axios.get(`/api/v1/stadiums/profile/${stadiumId}`);
+
+      setGallery(data.data.Stadium.stadiumGallery);
+      setUserData(data.data as UserData);
+    } catch (error) {
+      if ((error as errorI).response?.status === 401) {
+        navigate('/pageNotFound');
+      } else {
+        navigate('/serverError');
+      }
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Why you are using `useEffect` here, the page will be re rendered by default when its path changes, it will act as opening another page.